catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/AccessibilityPreferencesActivityTest.java

```diff
@@ -0,0 +1,156 @@
+		onView(withText(R.string.preference_access_title_fontface_dyslexic)).perform(click());
+		assertEquals(sharedPreferences.getString(ACCESS_MYPROFILE_FONTFACE, ""), ACCESS_FONTFACE_VALUE_DYSLEXIC);
+
+		onView(withText(R.string.close)).perform(click());
+
+		onView(withText(R.string.preference_access_title_large_text)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_TEXT, false));
+
+		onView(withText(R.string.preference_access_title_high_contrast)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_HIGH_CONTRAST, false));
+
+		onView(withText(R.string.preference_access_title_additional_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_ADDITIONAL_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_large_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_high_contrast_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_HIGH_CONTRAST_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_largeelementspacing)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_ELEMENT_SPACING, false));
+
+		onView(withText(R.string.preference_access_title_starter_bricks)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_STARTER_BRICKS, false));
+
+		onView(withText(R.string.preference_access_title_dragndrop_delay)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_DRAGNDROP_DELAY, false));
+
+		onView(withId(R.id.scratch_project_apply_button)).perform(click());
+		onView(withText(R.string.ok)).perform(click());
+
+		for (String setting : settings) {
+			assertTrue(sharedPreferences.getBoolean(setting, false));
+		}
+		assertEquals(sharedPreferences.getString(BaseSettingsActivity.ACCESS_FONTFACE, ""),
+				BaseSettingsActivity.ACCESS_FONTFACE_VALUE_DYSLEXIC);
+
+		openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getTargetContext());
+		onView(withText(R.string.settings)).perform(click());
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_access))
+				.perform(click());
+		onView(withId(R.id.access_switch_to_predefined_profiles_button)).perform(click());
+		onView(withText(R.string.preference_access_title_profile_standard)).perform(click());
+		onView(withId(R.id.scratch_project_apply_button)).perform(click());
+		onView(withText(R.string.ok)).perform(click());
+
+		for (String setting : settings) {
+			assertFalse(sharedPreferences.getBoolean(setting, false));
+		}
+		assertEquals(sharedPreferences.getString(BaseSettingsActivity.ACCESS_FONTFACE, ""),
+				BaseSettingsActivity.ACCESS_FONTFACE_VALUE_STANDARD);
+	}
+}
```

Comment:
i would rewrite this test to:

- save current state of shared preferences in setup
- set all to false in setup
- test if all options are available, and if they write into shared preferences correctly (and NOT apply / restart)
- reset the shared preference values to what they were before the test and restart